lib/livebook_web/live/session_live/secrets_component.ex

```diff
@@ -332,38 +337,43 @@ defmodule LivebookWeb.SessionLive.SecretsComponent do
     Livebook.Session.set_secret(socket.assigns.session.pid, secret)
   end
 
-  defp set_secret(_socket, %Secret{origin: {:hub, id}} = secret) when is_binary(id) do
-    Livebook.Hubs.create_secret(secret)
+  defp set_secret(socket, %Secret{origin: {:hub, id}} = secret) when is_binary(id) do
+    with :ok <- Livebook.Hubs.create_secret(secret) do
+      Livebook.Session.set_secret(socket.assigns.session.pid, secret)
+    end
   end
 
   defp grant_access(secrets, secret_name, origin, socket) do
     secret = Enum.find(secrets, &(&1.name == secret_name and &1.origin == origin))
 
-    if secret,
-      do: set_secret(socket, secret),
-      else: :ok
+    if secret, do: Livebook.Session.set_secret(socket.assigns.session.pid, secret)
   end
 
-  defp must_grant_access(%{assigns: %{prefill_secret_name: secret_name}} = socket) do
-    if not session?(socket, secret_name) and
-         (app?(socket, secret_name) or hub?(socket, secret_name)) do
-      secret_name
+  defp get_secret(socket, secret_name) do
+    session_secret = get_session_secret(socket, secret_name)
+    app_secret = get_app_secret(socket, secret_name)
+    hub_secret = get_hub_secret(socket, secret_name)
+
+    secrets = [session_secret, app_secret, hub_secret]
+
+    case Enum.reject(secrets, &is_nil/1) do
+      [] -> nil
+      [secret | _] -> secret
     end
```

Comment:
```suggestion
    get_app_secret(socket, secret_name) || get_hub_secret(socket, secret_name)
```

I don't think we should look at the session secret because the session is already granted.